libs/pika/execution/include/pika/execution/algorithms/drop_operation_state.hpp

```diff
@@ -248,5 +249,16 @@ namespace pika::execution::experimental {
         {
             return detail::partial_algorithm<drop_operation_state_t>{};
         }
-    } drop_operation_state{};
+    };
+
+    /// \brief Releases the operation state of the adaptor before signaling a connected receiver.
+    /// The operation_state of previous senders can potentially hold on to a allocated memory longer
+    /// than necessary which would prevent other algorithm from accessing this resource.
+    ///
+    /// Sender adaptor that takes any sender and returns a sender. Values sent as references will be
+    /// copied.
```

Comment:
Just checking: it just manipulate the operation state, but it proxies the sender that receives somehow? What I mean is, can I have something like this?

```cpp
ex::just(26) | ex::drop_operation_state() | ex::then([](int){});
```

Reply:
Yep, the above is possible. The

> Values sent as references will be copied.

sentence is meant to explain this aspect. Do you think that's clear enough? We can also consider adding another example demonstrating this.

Reply:
Probably being a bit more explicit won't harm.

> Sender adaptor that takes any sender and returns a sender with the same values received (except for values sent as references which will be copied).

Not sure it is correct, but I think you might get my point about it.

For what concerns the example, if you feel it would not distract too much, you might think about passing around a sender of something instead of a sender of void in the already existing example. At least for me it would help distinguish between the operation state and the sender and make clear that one is filtered and the other pass through.

Reply:
That's exactly what I was thinking! Thanks!

I wouldn't care about the "special case" of references. What I was targeting was exactly what you did, i.e. differentiating the operation state from the values sent through.